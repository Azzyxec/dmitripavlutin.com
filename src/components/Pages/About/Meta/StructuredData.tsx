import { Helmet } from 'react-helmet';

import { useAuthorAndSideInfo } from 'hooks/useAuthorAndSiteInfo';

export default function AboutMetaStructuredData() {
  const { author, site } = useAuthorAndSideInfo();
  const authorProfilePictureUrl = `${site.url}${author.profilePictureSrc}`;
  const sameAs = Object.keys(author.info.profiles)
    .reduce((list, key: keyof typeof author.info.profiles) => [...list, author.info.profiles[key]], []);
  const structuredData = {
    '@context': 'https://schema.org',
    '@type': 'Person',
    name: author.info.name,
    image: {
      '@type': 'ImageObject',
      url: authorProfilePictureUrl,
      width: 256,
      height: 256,
    },
    url: site.url,
    sameAs: sameAs,
    description: author.info.description,
    jobTitle: author.info.jobTitle,
    email: author.info.email,
  };
  return (
    <Helmet>
      <script type="application/ld+json">{JSON.stringify(structuredData, undefined, 2)}</script>
    </Helmet>
  );
}
