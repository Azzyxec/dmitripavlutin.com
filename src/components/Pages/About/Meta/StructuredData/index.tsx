import { Helmet } from 'react-helmet';

interface AboutMetaStructuredDataProps {
  siteInfo: SiteInfo;
  authorInfo: AuthorInfo;
  authorProfilePictureSrc: string;
}

export default function AboutMetaStructuredData({
  siteInfo,
  authorInfo,
  authorProfilePictureSrc,
}: AboutMetaStructuredDataProps) {
  const authorProfilePictureUrl = `${siteInfo.url}${authorProfilePictureSrc}`;
  const sameAs = Object.keys(authorInfo.profiles).reduce((list, key) => [...list, authorInfo.profiles[key]], []);
  const structuredData = {
    '@context': 'https://schema.org',
    '@type': 'Person',
    name: authorInfo.name,
    image: {
      '@type': 'ImageObject',
      url: authorProfilePictureUrl,
      width: 256,
      height: 256,
    },
    url: siteInfo.url,
    sameAs: sameAs,
    description: authorInfo.description,
    jobTitle: authorInfo.jobTitle,
    email: authorInfo.email,
  };
  return (
    <Helmet>
      <script type="application/ld+json">{JSON.stringify(structuredData, undefined, 2)}</script>
    </Helmet>
  );
}
