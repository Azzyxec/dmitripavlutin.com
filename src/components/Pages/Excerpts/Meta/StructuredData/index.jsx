import React from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';

export default function IndexMetaStructuredData(props) {
  const { data: { site: { siteMetadata }, authorProfilePicture } } = props;
  const authorProfilePictureUrl = `${siteMetadata.siteUrl}${authorProfilePicture.childImageSharp.resize.src}`;
  const structuredData = {
    "@context": "https://schema.org",
    "@type": "Website",
    "publisher": {
      "@type": "Organization",
      "name": siteMetadata.title,
      "logo": {
        "@type": "ImageObject",
        "url": authorProfilePictureUrl
      }
    },
    "url": siteMetadata.siteUrl,
    "image": {
      "@type": "ImageObject",
      "url": authorProfilePictureUrl,
      "width": 256,
      "height": 256
    },
    "description": siteMetadata.description,
    "mainEntityOfPage": {
      "@type": "WebPage",
      "@id": siteMetadata.siteUrl
    }
  };
  return (
    <Helmet>
      <script type="application/ld+json">
        {JSON.stringify(structuredData, undefined, 2)}
      </script>
    </Helmet>
  );
}

IndexMetaStructuredData.propTypes = {
  data: PropTypes.object
};