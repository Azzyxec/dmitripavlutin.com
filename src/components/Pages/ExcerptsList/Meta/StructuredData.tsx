import { Helmet } from 'react-helmet'
import { useSiteMetadata } from 'hooks/useSiteMetadata'

export function ExcerptsListMetaStructuredData() {
  const { author, site } = useSiteMetadata()

  const authorProfilePictureUrl = `${site.url}${author.profilePictureSrc}`
  const structuredData = {
    '@context': 'https://schema.org',
    '@type': 'Website',
    publisher: {
      '@type': 'Organization',
      name: site.metaTitle,
      logo: {
        '@type': 'ImageObject',
        url: authorProfilePictureUrl,
      },
    },
    url: site.url,
    image: {
      '@type': 'ImageObject',
      url: authorProfilePictureUrl,
      width: 256,
      height: 256,
    },
    description: site.description,
    mainEntityOfPage: {
      '@type': 'WebPage',
      '@id': site.url,
    },
  }
  return (
    <Helmet>
      <script type="application/ld+json">{JSON.stringify(structuredData, undefined, 2)}</script>
    </Helmet>
  )
}
