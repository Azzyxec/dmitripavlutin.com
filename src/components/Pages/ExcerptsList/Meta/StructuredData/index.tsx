import * as React from 'react';
import { Helmet } from 'react-helmet';

interface MetaStructuredData {
  siteInfo: SiteInfo;
  authorProfilePictureSrc: string;
}

export default function MetaStructuredData({ siteInfo, authorProfilePictureSrc }: MetaStructuredData) {
  const authorProfilePictureUrl = `${siteInfo.url}${authorProfilePictureSrc}`;
  const structuredData = {
    '@context': 'https://schema.org',
    '@type': 'Website',
    publisher: {
      '@type': 'Organization',
      name: siteInfo.metaTitle,
      logo: {
        '@type': 'ImageObject',
        url: authorProfilePictureUrl,
      },
    },
    url: siteInfo.url,
    image: {
      '@type': 'ImageObject',
      url: authorProfilePictureUrl,
      width: 256,
      height: 256,
    },
    description: siteInfo.description,
    mainEntityOfPage: {
      '@type': 'WebPage',
      '@id': siteInfo.url,
    },
  };
  return (
    <Helmet>
      <script type="application/ld+json">{JSON.stringify(structuredData, undefined, 2)}</script>
    </Helmet>
  );
}
