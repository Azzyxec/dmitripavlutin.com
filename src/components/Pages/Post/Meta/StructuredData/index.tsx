import React from 'react';
import Helmet from 'react-helmet';

interface PostMetaStructuredDataProps {
  post: Post;
  siteMetadata: SiteMetadata;
  authorProfilePicture: FluidImage
}

export default function PostMetaStructuredData({ post, siteMetadata, authorProfilePicture }: PostMetaStructuredDataProps) {
  const postUrl = `${siteMetadata.siteUrl}/${post.slug}/`;
  const postImageUrl = `${siteMetadata.siteUrl}${post.thumbnail.src}`;
  const authorProfilePictureUrl = `${siteMetadata.siteUrl}${authorProfilePicture.src}`;
  const sameAs = Object.keys(siteMetadata.profiles).reduce((sameAs, key) => [...sameAs, siteMetadata.profiles[key]], []);
  const structuredData = {
    "@context": "https://schema.org",
    "@type": "Article",
    "publisher": {
      "@type": "Organization",
      "name": siteMetadata.title,
      "logo": {
        "@type": "ImageObject",
        "url": authorProfilePictureUrl
      }
    },
    "author": {
      "@type": "Person",
      "name": siteMetadata.author,
      "image": {
        "@type": "ImageObject",
        "url": authorProfilePictureUrl,
        "width": 256,
        "height": 256
      },
      "url": siteMetadata.siteUrl,
      "sameAs": sameAs,
      "description": siteMetadata.speciality
    },
    "headline": post.title,
    "url": postUrl,
    "datePublished": post.published,
    "dateModified": post.modified,
    "image": {
      "@type": "ImageObject",
      "url": postImageUrl,
      "width": 720,
      "height": 400
    },
    "keywords": post.tags.join(', '),
    "description": post.description,
    "mainEntityOfPage": {
      "@type": "WebPage",
      "@id": siteMetadata.siteUrl
    }
  };
  return (
    <Helmet>
      <script type="application/ld+json">
        {JSON.stringify(structuredData, undefined, 2)}
      </script>
    </Helmet>
  );
}