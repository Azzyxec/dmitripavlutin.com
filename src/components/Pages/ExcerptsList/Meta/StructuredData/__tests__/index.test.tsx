import { shallow } from 'enzyme';

import StructuredData from '../index';

const siteInfo: SiteInfo = {
  title: 'Dmitri Pavlutin',
  description: 'Thoughts on Frontend development',
  metaTitle: 'Dmitri Pavlutin Blog',
  metaDescription: 'Posts by Dmitri Pavlutin about software development',
  url: 'https://dmitripavlutin.com',
  repositoryUrl: 'https://github.com/panzerdp/dmitripavlutin.com',
};

describe('<ExcerptsListStructuredData />', function() {
  it('should render structured data', function() {
    const wrapper = shallow(<StructuredData siteInfo={siteInfo} authorProfilePictureSrc="/image.png" />);
    expect(wrapper).toMatchSnapshot();
  });
});
