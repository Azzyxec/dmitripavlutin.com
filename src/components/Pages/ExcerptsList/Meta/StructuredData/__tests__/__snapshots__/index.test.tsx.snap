// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExcerptsListStructuredData /> should render structured data 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
>
  <script
    type="application/ld+json"
  >
    {
  "@context": "https://schema.org",
  "@type": "Website",
  "publisher": {
    "@type": "Organization",
    "name": "Dmitri Pavlutin blog",
    "logo": {
      "@type": "ImageObject",
      "url": "https://dmitripavlutin.com/image.png"
    }
  },
  "url": "https://dmitripavlutin.com",
  "image": {
    "@type": "ImageObject",
    "url": "https://dmitripavlutin.com/image.png",
    "width": 256,
    "height": 256
  },
  "description": "Posts by Dmitri Pavlutin about software development",
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "https://dmitripavlutin.com"
  }
}
  </script>
</HelmetWrapper>
`;
